catalog/transformer/catalog-transformer-csv-queryresponse/src/test/java/ddf/catalog/transformer/csv/CsvMetacardTransformerTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.catalog.transformer.csv;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.MetacardTypeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CsvMetacardTransformerTest {
+
+    private static Set<AttributeDescriptor> attributeDescriptorsSet;
+
+    private static List<AttributeDescriptor> attributeDescriptorsList;
+
+    private static Map<String, Attribute> metacardDataMap;
+
+    private static List<AttributeDescriptor> ATTRIBUTE_DESCRIPTOR_LIST;
+
+    Map<String, Serializable> arguments;
+
+    private CsvMetacardTransformer transformer;
+
+    Metacard normalMC;
+
+    private static final List<ImmutableTriple<Object, Object, Object>> ATTRIBUTE_DATA =
```

Comment:
The tuple seems the easiest way to contain all the data for the attributes we want since it can contain attribute name, value, and type. If there is already an existing object that contains all those I will use that instead.